index.mjs

```diff
@@ -297,10 +300,17 @@ export const coroutine = function coroutine(g) {
         );
       }
 
-      return done ? Parser.of(value) : value.chain(step);
```

Comment:
You're using `value.chain` here in the removed code, which I guess was the reason for the if statement prior to this explicitly checking for the existence of the chain method. But now, is there still a reason to do so? It seems like a fairly permissive check, as well. It seems like I could yield any value with a chain method.